plugins/ci/pkg/ci/git_test.go

```diff
@@ -0,0 +1,146 @@
+package ci
+
+import (
+	"errors"
+	"fmt"
+	"os/exec"
+	"testing"
+
+	"github.com/fairwindsops/insights-plugins/plugins/ci/pkg/models"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestExtractRepoNameFromOrigin(t *testing.T) {
+	assert.Equal(t, "FairwindsOps/insights-plugins", extractRepoNameFromOrigin("git@github.com:FairwindsOps/insights-plugins.git"))
+	assert.Equal(t, "", extractRepoNameFromOrigin(""))
+}
+
+func TestGetGitInfo(t *testing.T) {
+	r, err := getGitInfo(successfulStubExecutor, models.GithubActions, ".", "fairwinds/insights-plugins", "main")
+	assert.NoError(t, err)
+	assert.Equal(t, &gitInfo{origin: "origin-url", branch: "branch-name", masterHash: "master-hash-1", currentHash: "current-hash", commitMessage: "commit-message", repoName: "fairwinds/insights-plugins"}, r)
+
+	r, err = getGitInfo(successfulStubExecutorOriginBranch, models.GithubActions, ".", "fairwinds/insights-plugins", "main")
+	assert.NoError(t, err)
+	assert.Equal(t, &gitInfo{origin: "origin-url", branch: "branch-name", masterHash: "master-hash-2", currentHash: "current-hash", commitMessage: "commit-message", repoName: "fairwinds/insights-plugins"}, r)
+
+	r, err = getGitInfo(successfulStubExecutorRemotesOriginBranch, models.GithubActions, ".", "fairwinds/insights-plugins", "main")
+	assert.NoError(t, err)
+	assert.Equal(t, &gitInfo{origin: "origin-url", branch: "branch-name", masterHash: "master-hash-3", currentHash: "current-hash", commitMessage: "commit-message", repoName: "fairwinds/insights-plugins"}, r)
+
+	r, err = getGitInfo(errorOnOptionalStubExecutor, models.GithubActions, ".", "fairwinds/insights-plugins", "main")
+	assert.NoError(t, err)
+	assert.Equal(t, &gitInfo{origin: "", branch: "", masterHash: "", currentHash: "current-hash", commitMessage: "", repoName: "fairwinds/insights-plugins"}, r)
+
+	r, err = getGitInfo(errorOnRequiredStubExecutor, models.GithubActions, ".", "fairwinds/insights-plugins", "main")
+	assert.Error(t, err)
+	assert.Nil(t, r)
+}
+
```

Comment:
I like how you've abstracted the executor function to test `getGitInfo`. I do wish the helper functions for tests felt easier for me to follow, but I can't think of how they could be improved at the moment.

Reply:
Yea, I thought of having the return based on how many times the function has been called but sounded like a very obscure approach